cmd/gapir/cc/main.cpp

```diff
@@ -144,7 +145,28 @@ void android_main(struct android_app* app) {
 
     // Note if you want to create a disk cache create it under:
     // app->activity->internalDataPath
-    listenConnections(std::move(conn), nullptr, nullptr, Connection::NO_TIMEOUT, &memoryManager, crashHandler);
+    std::thread listening_thread([&]() {
+      listenConnections(conn.get(), nullptr, nullptr, Connection::NO_TIMEOUT,
+                        &memoryManager, crashHandler);
+    });
+    while (true) {
+      int ident;
+      int fdesc;
+      int events;
+      struct android_poll_source* source;
+      while ((ident = ALooper_pollAll(0, &fdesc, &events, (void**)&source)) >=
+             0) {
+        // process this event
+        if (source) {
+          source->process(app, source);
+        }
+        if (app->destroyRequested) {
+          conn->close();
```

Comment:
Is this the only code that requires changing the Connection from a unique_ptr to a raw pointer? If so would it be possible to find another way of closing the connection? Something like returning an interrupt lambda from the ServerListener inside listenConnections may work. Either that, or perhaps get rid of the unique_ptr entirely and manually allocate/free the Connection. That way the ownership and destruction process for the Connection memory will be more clear.

@ben-clayton, thoughts?

Reply:
Passing around pointers from unique_ptr objects is a very common operation. 
All this does it stop transferring ownership of the connection into listenConnections. I am very much against getting rid of the unique_ptr, since it's creation/destruction is well-defined here. It is always destroyed at the end of the main() function. It is closed earlier, but the actual connection object remains valid beyond that point.

Reply:
I feel the same as @AWoloszyn , the ownership is quite well defined here, so I think passing a raw pointer should be fine?

Reply:
At a previous company I worked we had a policy of using a reference instead of a pointer if you can ensure that the lifetime of the object is guaranteed for the duration of the call.
Not sure if this aligns with Google idiomatic guides.

Reply:
I thought Google C++ style says that, if the object is to be used as const, a const reference is preferred, if it may be modified, pass it with a pointer.

Reply:
Yeah, the Google style-guide basically says "No non-const references anywhere ever, unless you are doing it to be consistent with the STL"